pkg/clusteragent/autoscaling/cluster/config_retriever.go

```diff
@@ -0,0 +1,73 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package cluster
+
+import (
+	"context"
+
+	"k8s.io/utils/clock"
+
+	"github.com/DataDog/datadog-agent/pkg/config/remote/data"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const configRetrieverStoreID string = "cacr"
+
+// RcClient is a subinterface of rcclient.Component to allow mocking
+type RcClient interface {
+	SubscribeIgnoreExpiration(product string, fn func(update map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)))
+}
+
+// ConfigRetriever is responsible for retrieving remote objects (Cluster Autoscaling values)
+type ConfigRetriever struct {
+	isLeader func() bool
+	clock    clock.WithTicker
```

Comment:
It seems like you're not using the clock. I believe you do need it for the async apply flow (in case an update was lost due to leader change)